lib/ipam/ipam_block_reader_writer.go

```diff
@@ -377,18 +380,31 @@ func blockGenerator(pool cnet.IPNet) func() *cnet.IPNet {
 // Returns a generator that, when called, returns a random
 // block from the given pool.  When there are no blocks left,
 // the it returns nil.
-func randomBlockGenerator(pool cnet.IPNet, hostName string) func() *cnet.IPNet {
+func randomBlockGenerator(ipPool *v3.IPPool, hostName string) func() *cnet.IPNet {
+	_, pool, err := cnet.ParseCIDR(ipPool.Spec.CIDR)
+	if err != nil {
```

Comment:
ick, I guess we should log an error in this case so if somehow we do get here we'll see it in the logs.